includes/sanitizers/class-amp-o2-player-sanitizer.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Class AMP_O2_Player_Sanitizer
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_O2_Player_Sanitizer
+ *
+ * Converts <div class="vdb_player><script></script></div> embed to <amp-o2-player>
+ *
+ * @see https://www.ampproject.org/docs/reference/components/amp-o2-player
+ */
+class AMP_O2_Player_Sanitizer extends AMP_Base_Sanitizer {
+	const URL_PATTERN = '#.*delivery.vidible.tv\/jsonp\/pid=(.*)\/vid=(.*)\/(.*).js.*#i';
+
+	/**
+	 * AMP Tag.
+	 *
+	 * @var string AMP Tag.
+	 */
+	public static $amp_tag = 'amp-o2-player';
+
+	/**
+	 * Amp O2 Player class.
+	 *
+	 * @var string CSS class to identify O2 Player <div> to replace with AMP version.
+	 */
+	public static $xpath_selector = '//div[ contains( @class, \'vdb_player\' ) ]/script';
+
+	/**
+	 * Height to set for 02 Player elements.
+	 *
+	 * @var string
+	 */
+	private static $height = '270';
+
+	/**
+	 * Width to set for 02 Player elements.
+	 *
+	 * @var string
+	 */
+	private static $width = '480';
+
+	/**
+	 * XPath.
+	 *
+	 * @var DOMXPath
+	 */
+	private $xpath;
+
+	/**
+	 * AMP_O2_Player_Sanitizer constructor.
+	 *
+	 * @param DOMDocument $dom  Represents the HTML document to sanitize.
+	 * @param array       $args Args.
+	 */
+	public function __construct( DOMDocument $dom, array $args = array() ) {
+		parent::__construct( $dom, $args );
+		$this->xpath = new DOMXPath( $dom );
```

Comment:
The `xpath` setting here could just as well be moved to the `sanitize` method since that is where it is used. This would avoid needing to override the constructor at all, and if the sanitizers are instantiated but never call `sanitize` then we could avoid needlessly instantiating this object (however rare that may be).